### PR TITLE
feat(headercolor): add HeaderColor plugin

### PR DESCRIPTION
**Pull request by John-Luke**
fix [#760](https://github.com/driftyco/ionic-native/issues/760)

### CODE REVIEW

File: src/plugins/headercolor.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+})
+export class HeaderColor {
+
+  /**
+   * Set a color to the task header
+   * @param color {string} The color
+   * @param {Object} Optional callbacks
+   */
+  @Cordova()
+  static tint(color: string, callbacks?: {success?: Function, failure?: Function}): void { }
```
Comment (ihadeed) [inline src/plugins/headercolor.ts]: This should return a Promise.

Also the decorator should include the following options:
```
@Cordova({
  callbackStyle: 'object',
  successName: 'success',
  errorName: 'failure'
})
```

**Comment by John-Luke** [comment]
@ihadeed

File: src/plugins/headercolor.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  /**
+   * Set a color to the task header
+   * @param color {string} The color 
+   */
+  @Cordova({
+    callbackStyle: 'object',
+    successName: 'success',
+    errorName: 'failure'
+  })
+  static tint(color: string): Promise<any> { }
```
Comment (ihadeed) [inline src/plugins/headercolor.ts]: Need to add `return; ` between the curly brackets to avoid TS Errors.

File: src/plugins/headercolor.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  pluginRef: 'headercolor',
+  repo: 'https://github.com/tomloprod/cordova-plugin-headercolor',
+  platforms: ['Android']
+})
+export class HeaderColor {
+
+  /**
+   * Set a color to the task header
+   * @param color {string} The color 
+   */
```
Comment (ihadeed) [inline src/plugins/headercolor.ts]: add `@returns {Promise<any>}`

**Comment by John-Luke** [comment]
@ihadeed

File: src/plugins/headercolor.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * ```typescript
+ * import { HeaderColor } from 'ionic-native';
+ *
+ * HeaderColor.tint("#becb29");
+ * ```
+ */
+@Plugin({
+  name: 'HeaderColor',
+  plugin: 'cordova-plugin-headercolor',
+  pluginRef: 'headercolor',
```
Comment (mebibou) [inline src/plugins/headercolor.ts]: Shouldn't this be `pluginRef: 'headerColor',` ?

Comment (ihadeed) [inline src/plugins/headercolor.ts]: just had a look

It should be `plugins.headerColor`

**Comment by ihadeed** [comment]
@John-Luke changes look good, please see https://github.com/driftyco/ionic-native/pull/821#discussion_r90491288

**Pull request merged**